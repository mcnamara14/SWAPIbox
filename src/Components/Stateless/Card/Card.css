h2 {
  font-size: 1.5rem; }

h3 {
  font-size: 1rem; }

.card {
  background: #222;
  width: 23%;
  padding: 20px 0;
  color: #fff;
  margin: 10px;
  display: inline-block; }

.cards {
  display: flex;
  flex-wrap: wrap; }

.favorite {
  background-image: url("./star-inactive.png");
  background-size: contain;
  background-repeat: no-repeat;
  background-position-x: 90%;
  width: 100%;
  height: 35px;
  display: block; }
