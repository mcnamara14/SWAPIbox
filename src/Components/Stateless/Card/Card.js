import React from 'react';
import './Card.css';

const Card = ({ data, id, findCard, selected }) => {
  const keys = Object.keys(data);
  return (
    <article className={selected ? "card favorite" : "card"}>

        { keys.map((key, index) => {
          if (key === 'name') {
            return (
            <div className="cardHeader" key={index}>
              <h2>{data.name}</h2>      
              <div className="cardFave">
                <a className="favoriteIcon" onClick={() => findCard(data.id)}></a>
              </div>
            </div>
            )
          } else if (key === 'id' || key === 'favorite'){
            return null
          } else {
            return <p><span>{`${key}:`}</span> {data[key]}</p>
          }
            }
        )}
        <div className="cardBottom"></div>
    </article>
  )
};

export default Card;

