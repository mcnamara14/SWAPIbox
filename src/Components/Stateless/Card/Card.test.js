import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      data: {
        name: "C-3PO", 
        homeworld: "Tatooine", 
        species: "Droid", 
        population: "200000", 
        id: "people1"
      },
      id: 'people1',
      findCard: jest.fn(),
      selected: { card: 'data' },
    }
    wrapper = shallow(<Card {...mockProps} />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should add favorite class if a card is selected', () => {
    expect(wrapper.find('.favorite').length).toEqual(1);
  })

  it('should not add favorite class if a card is not selected', () => {
    mockProps = {
      data: {
        name: "C-3PO", 
        homeworld: "Tatooine", 
        species: "Droid", 
        population: "200000", 
        id: "people1"
      },
      id: 'people1',
      findCard: jest.fn(),
      selected: null,
    }

    wrapper = shallow(<Card {...mockProps} />)

    expect(wrapper.find('.favorite').length).toEqual(0);
  })

  it('should call findCard when favoriteIcon is clicked', () => {
    const mockFunction = jest.fn(); 
    const wrapper = shallow(<Card {...mockProps} findCard={mockFunction} />)
    
    wrapper.find('.favoriteIcon').simulate('click');

    expect(mockFunction).toHaveBeenCalled()  
  })
})
