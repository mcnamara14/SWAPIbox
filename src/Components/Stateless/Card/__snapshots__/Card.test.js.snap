// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Object {
            "homeworld": "Tatooine",
            "id": "people1",
            "name": "C-3PO",
            "population": "200000",
            "species": "Droid",
          }
    }
    findCard={[Function]}
    id="people1"
    selected={
        Object {
            "card": "data",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <div
            className="cardHeader"
>
            <h2>
                        C-3PO
            </h2>
            <div
                        className="cardFave"
            >
                        <a
                                    className="favoriteIcon"
                                    onClick={[Function]}
                        />
            </div>
</div>,
          <p>
            <span>
                        homeworld:
            </span>
             
            Tatooine
</p>,
          <p>
            <span>
                        species:
            </span>
             
            Droid
</p>,
          <p>
            <span>
                        population:
            </span>
             
            200000
</p>,
          null,
        ],
        <div
          className="cardBottom"
/>,
      ],
      "className": "card favorite",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              C-3PO
</h2>,
            <div
              className="cardFave"
>
              <a
                            className="favoriteIcon"
                            onClick={[Function]}
              />
</div>,
          ],
          "className": "cardHeader",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "C-3PO",
            },
            "ref": null,
            "rendered": "C-3PO",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="favoriteIcon"
                onClick={[Function]}
/>,
              "className": "cardFave",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "favoriteIcon",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              homeworld:
</span>,
            " ",
            "Tatooine",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "homeworld:",
            },
            "ref": null,
            "rendered": "homeworld:",
            "type": "span",
          },
          " ",
          "Tatooine",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              species:
</span>,
            " ",
            "Droid",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "species:",
            },
            "ref": null,
            "rendered": "species:",
            "type": "span",
          },
          " ",
          "Droid",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              population:
</span>,
            " ",
            "200000",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "population:",
            },
            "ref": null,
            "rendered": "population:",
            "type": "span",
          },
          " ",
          "200000",
        ],
        "type": "p",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "cardBottom",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <div
              className="cardHeader"
>
              <h2>
                            C-3PO
              </h2>
              <div
                            className="cardFave"
              >
                            <a
                                          className="favoriteIcon"
                                          onClick={[Function]}
                            />
              </div>
</div>,
            <p>
              <span>
                            homeworld:
              </span>
               
              Tatooine
</p>,
            <p>
              <span>
                            species:
              </span>
               
              Droid
</p>,
            <p>
              <span>
                            population:
              </span>
               
              200000
</p>,
            null,
          ],
          <div
            className="cardBottom"
/>,
        ],
        "className": "card favorite",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                C-3PO
</h2>,
              <div
                className="cardFave"
>
                <a
                                className="favoriteIcon"
                                onClick={[Function]}
                />
</div>,
            ],
            "className": "cardHeader",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "C-3PO",
              },
              "ref": null,
              "rendered": "C-3PO",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="favoriteIcon"
                  onClick={[Function]}
/>,
                "className": "cardFave",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "favoriteIcon",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "a",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                homeworld:
</span>,
              " ",
              "Tatooine",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "homeworld:",
              },
              "ref": null,
              "rendered": "homeworld:",
              "type": "span",
            },
            " ",
            "Tatooine",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                species:
</span>,
              " ",
              "Droid",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "species:",
              },
              "ref": null,
              "rendered": "species:",
              "type": "span",
            },
            " ",
            "Droid",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                population:
</span>,
              " ",
              "200000",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "population:",
              },
              "ref": null,
              "rendered": "population:",
              "type": "span",
            },
            " ",
            "200000",
          ],
          "type": "p",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "cardBottom",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
